src/main/java/org/apache/sling/jcr/resource/internal/helper/jcr/JcrResourceProvider.java

```diff
@@ -358,8 +358,8 @@ public Resource getResource(ResolveContext<JcrProviderState> ctx, String path, R
                 if (version == null) {
                     String parentPath = ResourceUtil.getParent(child.getPath());
                     if (parentPath != null) {
-                        Item parentItem = ctx.getProviderState().getResourceFactory()
-                            .getItemOrNull(parentPath);
+                        Item parentItem = ((JcrItemResource)child).getItem().getParent();
```

Comment:
getItem() as well as getParent might throw an exception - and will not return null, so I think the existing code is correct